CanOpen-and-Canfestivino-installation-guide.md

```diff
@@ -0,0 +1,302 @@
+* Jumper cables (Male-male, male-female, female-female)
+* Potentiometer
+* Servo (SG90)
+* Breadboard
+
+*) The implementation is not limited to this specific hardware and software, but was not tested with other systems. It might be possible to eg. use Windows Subsystem for Linux (WSL), or other CANBUS interfaces such as the PiCAN shield.
+
+## Wiring guide
+As the Raspberry Pi GPIO pins are only rated for 3.3V, it's important to wire the SBC-CAN01 correctly. The SBC-CAN01 is designed in such a way that the SPI interface is powered via the VCC pin, and thus uses 3.3V for SPI.
+However, the CAN transceiver chip requires 5V, which is supplied via the VCC1 pin. See the datasheet for the SBC-CAN01 for more details.
+
+On the Arduino Uno, it's okay to supply 5V to both VCC and VCC1.
+
+\#TODO add wiring diagram
+
+## Raspberry Setup
+###CANBUS Setup
+
+To install the necessary extensions and to be able to use the CAN module with a Raspberry Pi using CanOpen, please update the package list and install the extensions can-ultis:
+
+```
+sudo apt-get update
+sudo apt-get upgrade
+sudo apt-get install can-utils
+```
+
+Next, use the following command to get the kernel version of your Raspberry Pis. This is necessary for the further configuration of the system.
+```
+uname -a
+```
+You will be shown the current kernel version (e.g. "4.4.41-v7").
+
+Next, open and edit the `config.txt` file with the following command:
+
+```
+sudo nano /boot/config.txt
+```
+If your kernel version is 4.4.x or newer, please add the following lines to the end of the file:
+
+```
+dtparam=spi=on
+dtoverlay=mcp2515-can0,oscillator=16000000,interrupt=25
+dtoverlay=sp1-1cs
+```
+However, if your Raspberry Pi runs with an older kernel version, please add the following lines to the end of the file instead:
+
+```
+dtparam=spi=on
+dtoverlay=mcp2515-can0,oscillator=16000000,interrupt=25
+dtoverlay=sp1-bcm2835-overlay
+```
+
+The `16000000` stands that the MCP2515 operates with a clock of 16 MHz. This depends on the quartz used and may need to be adjusted.
 The SBC-CAN01 uses a 16MHz clock, so you can leave this unchanged.
+ 
Save the file with the key combination `CTRL+O`, confirm the saving process with `Enter`, and exit the editor with the combination `CTRL+X`.
+Restart your Raspberry Pi with the following command:
+
+```
+sudo reboot
+```
+
+After the restart has been completed, you can now start the CAN interface:
+
+```
+ sudo ip link set can0 up type can bitrate 500000
+```
+If you get errors saying that the buffer is full, extend it using the following command:
+
+```
+sudo ifconfig can0 txqueuelen 10000
+```
+
+The system should now be ready. You can now start a first send attempt. First, open a *new* terminal window and start candump with the following command:
+
+```
+candump can0
+```
+Then in the *other* terminal, run the following command:
+
+```
+cansend can0 127#DEADBEEF
+```
+In the windown running candump, you should see something similar to the following:
```

Comment:
'In the windown running candump, you' -> 'In the window with candump running, you '